### PR TITLE
Snapshotter

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

File: src/Kernel.php [hunk 1]
```diff
@@ -145,9 +148,15 @@ function pipleline(callable $firstCallback, callable ...$callbacks): callable
...
+function loadState(SnapshotStore $snapshotStore, Message $message, AggregateDefiniton $definiton): array
 {
-    return []; // @todo: fetch from projections
+    $aggregate = $snapshotStore->get($definiton->aggregateType(), $definiton->extractAggregateId($message));
+
+    if (! $aggregate) {
+        return [];
+    }
+
+    return $aggregate->aggregateRoot();
```
Comment (codeliner) [inline src/Kernel.php:159]: @prolic Do you think we can rename `Aggregate::aggregateRoot()` to `Aggregate::state()`?

**Review by codeliner — APPROVED** [review]

**Comment by codeliner** [comment]
My question is not related to this PR so I'm merging it.

File: src/Kernel.php (hunk 1 above)
Comment (prolic) [reply src/Kernel.php:159]: The snapshot-store is also used in the event-sourcing lib. I could not come up with a name that fits both perfectly.

**Review by basz — COMMENTED** [review]
i see many inline fqcn's. (not imported) but am guessing that's on purpose.

**Comment by prolic** [comment]
Where? Can you submit a pr?

Am 21.01.2017 23:02 schrieb "Bas Kamer" <notifications@github.com>:

**Pull request merged**